osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -136,39 +136,42 @@ private void createSsldPendingFirmwareUpdateRecord(final CorrelationIds ids, fin
                     .findByFirmwareFile(firmwareFile);
             final String firmwareVersion = firmwareFileFirmwareModule.getModuleVersion();
 
-            final FirmwareModuleType firmwareModuleType = firmwareModuleData.getFirmwareModuleType();
+            final Map<FirmwareModule, String> versionsByModule = firmwareModuleData
```

Comment:
Consider refactoring this method (maybe extract one or more methods) to increase readability.
You could possibly also find the needed firmware version and module type via FirmwareFile -> FirmwareFileFirmwareModules -> FirmwareModule?